lib/index.js

```diff
@@ -533,8 +546,6 @@ Socket.prototype._flush = function() {
 
         throw sendError;
       }
-
-      flags = this._ioevents;
```

Comment:
I moved this to the top of the send-loop, as it'll be reset on line 504 and if *anything*, it might have been effected by our last message reception (I have nothing to back this up, but it seems safer, if anything).